### PR TITLE
Google Login

### PR DESCRIPTION
**Pull request by leonplondon**
# Summary
Implemented Log in with Google.
Added dependencies.
Refactor of Login screen.
Implemented business logic.

# Trello Card
https://trello.com/c/vmm3eYIG/36-google-login

# Screenshots
screenshot_1544047842
screenshot_1544047847

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 1]
```diff
@@ -167,4 +202,24 @@ void onLoginClicked() {
...
+        }
+    }
+
+    public void handleSignInWithGoogleResult(Task<GoogleSignInAccount> task) {
+        try {
+            GoogleSignInAccount account = task.getResult(ApiException.class);
+            mToastFactory.showLong(getString(R.string.login_user_signing_with_google_account) + " " + account.getEmail());
+            onUserLoggedIn();
+        } catch (ApiException e) {
+            mToastFactory.showLong(String.format("%s %s", getString(R.string.app_unexpected_error), e.getMessage()));
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: We can save %s %s on strings file such as app_unexpected_error_format. What do you  think?

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 2]
```diff
@@ -167,4 +202,24 @@ void onLoginClicked() {
...
+            Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+            showProgress();
+            startActivityForResult(signInIntent, GOOGLE_SIGN_IN);
+        }
+    }
+
+    public void handleSignInWithGoogleResult(Task<GoogleSignInAccount> task) {
+        try {
+            GoogleSignInAccount account = task.getResult(ApiException.class);
+            mToastFactory.showLong(getString(R.string.login_user_signing_with_google_account) + " " + account.getEmail());
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: We can use here some format to avoid " "

**Review by lmiotti — CHANGES_REQUESTED** [review]
This is very good. Keep on going like this :+1: 
One question, have you created a Firebase account to test it?

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java (hunk 1 above)
Comment (leonplondon) [reply app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: My bad, I think it's good to save them on strings. Checking arguments of `getString`, I can simply put getString(id_string, ...args) and do the same

**Review by dylan-muszel — APPROVED** [review]
It's great 😄

**Review by lmiotti — APPROVED** [review]

**Pull request merged**